### PR TITLE
Make search of pending more specific

### LINKED ISSUE #369: Approving button is not updated
After the approval was mined, the approval was not updated and I could nto proceed with placing an order on mainnet: 
image

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #369 

This is an idea taken from the original repository, seems like when a stuck tx is found `useHasPendingApproval` returns true (this also returns false when the amount is not specified), I'm not completely sure that this was the problem in mainnet but being more common to have stuck tx in mainnet I would give a try to this approach.

### CODE REVIEW

**Review by mariano-aguero — APPROVED** [review]

File: src/state/transactions/hooks.tsx [hunk 1]
```diff
@@ -72,16 +72,31 @@ export function useAllTransactions(): { [txHash: string]: TransactionDetails } {
...
+      Object.keys(allTransactions).some((hash) => {
+        const tx = allTransactions[hash]
+        if (!tx) return false
+        if (tx.receipt) {
           return false
         } else {
-          return allTransactions[hash]?.approvalOfToken === tokenAddress
+          const { approvalOfToken } = tx
+          if (!approvalOfToken) return false
+          return approvalOfToken === tokenAddress && isTransactionRecent(tx)
```
Comment (josojo) [inline src/state/transactions/hooks.tsx:97]: Can you explain this? You are adding another constraint , so your funciton is returning less likely a true.

But we have the problem that it is not returning true, although it should return true. Hence, I think that your fix will not improve the situation.

Comment (hpmaxi) [reply src/state/transactions/hooks.tsx:97]: We should make that a user has a pending approval less likely to be true. Maybe I'm missing something, but I will re-check everything.

Comment (josojo) [reply src/state/transactions/hooks.tsx:97]: Ah I see.

Comment (josojo) [reply src/state/transactions/hooks.tsx:97]: Also the spender could be the problem: 
https://github.com/Uniswap/uniswap-interface/blob/8fd894f2d1b03921c5519c050aed164343c47fb1/src/state/transactions/hooks.tsx#L80

Maybe we should use uniswaps new logic for it...

**Comment by josojo** [comment]
FYI: 
we also have the current hack in the code:
```
// eslint-disable-next-line no-warning-comments
  // Todo: research why this hack of allowance reload for each block
  // is needed to update the allowance after approving.
  const allowance = useSingleCallResult(contract, 'allowance', inputs, {
    blocksPerFetch: 1,
  }).result
  ```
  
 Uniswap just uses:
 ```
 
  const allowance = useSingleCallResult(contract, 'allowance', inputs).result
  
  
```

Maybe its also related to that....

**Review by josojo — APPROVED** [review]
I wasn't able to reproduce tthis bug on mainnet. I think we can merge it like that.

**Pull request merged**